### PR TITLE
feat(freebsd): add FreeBSD support

### PR DESCRIPTION
**Pull request by Strade28**

### CODE REVIEW

**Comment by dafyddj** [comment]
The changes seem fine to me, but could do with being tested, no? (By which I mean adding a platform to test-kitchen)

**Comment by Strade27** [comment]
Hello, unfortunetly travis doesn’t support freebsd. I not found easy way to do that.

**Comment by dafyddj** [comment]
You could add a custom kitchen.yml, and leave it out from Travis. A manual test is better than no test, IMO.

**Comment by myoung34** [comment]
Manual tests are generally fine but they wont be run because maintainers just rely on the GH status API.

If the tests that are in place work, then the code is fine as-is, and if it works with FreeBSD then thats a bonus until travis supports that platform. Until then this is just a refactor change and freebsd is unofficially supported as far as tests/assertions are concerned

**Comment by Strade28** [comment]
Hello !
I add some local test for freebsd using kitchen-vagrant.

File: docs/README.rst [hunk 1]
```diff
@@ -32,7 +32,7 @@ which contains the currently released version. This formula is versioned accordi
 
 See `Formula Versioning Section <https://docs.saltstack.com/en/latest/topics/development/conventions/formulas.html#versioning>`_ for more details.
 
-Contributing to this repo
+Caontributing to this repo
```
Comment (n-rodriguez) [inline docs/README.rst]: Typo

File: .kitchen.freebsd.yml [hunk 2]
```diff
@@ -0,0 +1,87 @@
+---
+driver:
+  name: vagrant
+  hostname: freebsd.ci.local
+  cache_directory: false
+  customize:
+    memory: 512
+  provision: true
+  vagrantfiles:
+          - .freebsd_provision.rb
```
Comment (n-rodriguez) [inline .kitchen.freebsd.yml]: Indentation

**Comment by rbjorklin** [comment]
Just a heads up that my change that just got merged might have broken this.
https://github.com/saltstack-formulas/vault-formula/pull/33

**Comment by Strade28** [comment]
> Just a heads up that my change that just got merged might have broken this.
> #33

Thanks ! there is not problem with your change.

File: docs/README.rst [hunk 3]
```diff
@@ -95,13 +95,22 @@ Requirements
...
   gem install bundler
   bundle install
   bundle exec kitchen test all
 
+FreeBSD:
+::
+
+  gem install bundler
+  bundle install
+  bundle exec KITCHEN_LOCAL_YAML=.kitchen.freebsd.yml kitchen test -t test_freebsd/integration
```
Comment (n-rodriguez) [inline docs/README.rst]: This command doesn't work : `bundler: command not found: KITCHEN_LOCAL_YAML=.kitchen.freebsd.yml`
It should be `KITCHEN_LOCAL_YAML=.kitchen.freebsd.yml bundle exec kitchen test -t test_freebsd/integration`

**Comment by n-rodriguez** [comment]
there is an issue in dev-server-freebsd-11 converge step :

[dev-server-freebsd-11.log](https://github.com/saltstack-formulas/vault-formula/files/3320913/dev-server-freebsd-11.log)

**Comment by myii** [comment]
@Strade28 The following is a suggestion, not a requirement:

* This PR is ripe for taking advantage of our Cirrus CI integration, which supports FreeBSD.
* Examples to refer to include:
  - https://github.com/saltstack-formulas/rkhunter-formula/blob/master/.cirrus.yml
  - https://github.com/saltstack-formulas/syslog-ng-formula/blob/master/.cirrus.yml
* These can run concurrently with Travis CI, as done in the two formulas above.

Basically, leave Travis CI as-is but add a single entry for FreeBSD in Cirrus CI.

**Comment by fzipi** [comment]
@Strade28 Can you solve the conflicts so this can be merged?

**Comment by aboe76** [comment]
@dafyddj try this page: https://github.com/orgs/saltstack-formulas/teams

**Comment by aboe76** [comment]
@UtahDave can you invite @dafyddj ?

**Comment by myii** [comment]
@aboe76 How about this?

* https://help.github.com/en/articles/inviting-users-to-join-your-organization

**Comment by aboe76** [comment]
@myii yes I know but I don't have access, I'm only a moderator.

**Comment by myii** [comment]
@Strade28 We've finally got FreeBSD Vagrant testing working in GitHub Actions but unfortunately that's come a bit too late for this PR -- I wasn't able to push the changes here since your fork is no longer available.  However, I've recovered your excellent contribution and added the CI on top in #54.  I had to squash your commits and do a bit of jiggling but the backbone of your work is all there.

Thanks to all of the others for their comments and reviews towards this PR.

Closing this PR since nothing else can be done here.  Please continue any further discussion in #54.

**Comment by myii** [comment]
@Strade28 #54 has been merged, so the work in this PR wasn't lost in the end -- thanks again for the contribution.
